README.md

```diff
@@ -1,15 +1,89 @@
+```
+
+Below is an example integration test [from Jubilant itself](https://github.com/canonical/jubilant/blob/main/tests/integration/test_basic.py). This test uses the [`juju` pytest fixture](https://github.com/canonical/jubilant/blob/main/tests/integration/conftest.py), which adds a model during setup, runs your test with a `Juju` instance pointing at that model, and destroys the model during teardown:
+
+```python
+def test_deploy(juju: jubilant.Juju):
+    juju.deploy('snappass-test')             # Deploy the charm
+    status = juju.wait(jubilant.all_active)  # Wait till the app and unit are 'active'
+
+    # Hit the Snappass HTTP endpoint to ensure it's up and running.
+    address = status.apps['snappass-test'].units['snappass-test/0'].address
+    response = requests.get(f'http://{address}:5000/', timeout=10)
+    response.raise_for_status()
+    assert 'snappass' in response.text.lower()
+```
+
+You don't have to use [Pytest](https://docs.pytest.org/en/stable/) with Jubilant, but it's what we usually recommend. Pytest's fixtures and its simple `assert`-based approach make writing tests easy.
 
 
 ## Design goals
 
-- Match the Juju CLI. We try to ensure methods, argument names, and response field names match the Juju CLI and its responses, with minor exceptions (such as "application" being shortened to "app" in `Status` fields).
-- Simple API. Higher-level operations will be in helpers and fixtures, not the main `Juju` class.
-- No `async`. This was a "feature" of python-libjuju that isn't desired for integration tests.
+We designed Jubilant so it would:
+
+- Match the Juju CLI. Method, parameter, and response field names match the Juju CLI, with minor exceptions (such as "application" being shortened to "app").
+- Have a simple API. Higher-level operations will be in helpers and fixtures, not the main `Juju` class (the only exception being `Juju.wait`).
+- Not use `async`. This was a "feature" of python-libjuju that adds complexity and isn't needed for integration tests.
 - Support Juju 3 and 4. The Juju team is guaranteeing CLI arguments and `--format=json` responses won't change between Juju 3.x and 4.x. When Juju 5.x arrives and changes the CLI, we'll keep the Jubilant API simple and 1:1 with the 5.x CLI, but will consider adding a `jubilant.compat` layer to avoid tests have to manually work around differences between 4.x and 5.x.
+
+
+## Contributing and developing
+
+Anyone can contribute to Jubilant. It's best to start by [opening an issue](https://github.com/canonical/jubilant/issues) with a clear description of the problem or feature request, but you can also [open a pull request](https://github.com/canonical/jubilant/pulls) directly.
+
+Jubilant uses [`uv`](https://docs.astral.sh/uv/) to manage Python dependencies and tools, so you'll need to [install uv](https://docs.astral.sh/uv/#installation) to work on the library. You'll also need `make` to run local development tasks (but you probably have `make` installed already).
```

Comment:
You're almost certainly going to want to have Juju, I think. Do you feel that the link at the top to `juju.is` is sufficient for that? It does have a prominent link to the tutorial. I wonder if a link to [setting up a dev environment](https://canonical-juju.readthedocs-hosted.com/en/latest/user/howto/manage-your-deployment/manage-your-deployment-environment/#manage-your-deployment-environment) is worthwhile here as a helper? Or even something along the lines of "you can get a simple Juju environment using the 'charm-dev' Multipass blueprint: run `multipass launch ... charm-dev`".

Or is this all pointless because why are you looking at the Jubilant README if you don't already have Juju?

Reply:
True. I think this does somewhat go without saying. However, you also need it to *use* Jubilant, so I've added a paragraph with a link to the "install juju" page further up. They can follow further docs there if they really are new to Juju (which is highly unlikely if they want to use Jubilant).